From b9651c1e4aa70a62bbf2fdc589fc6528cc13f3bc Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Mon, 1 Sep 2014 09:25:11 -0400
Subject: [md] raid5: fix stripe release order

Message-id: <1409563584-9000-10-git-send-email-Jes.Sorensen@redhat.com>
Patchwork-id: 89948
O-Subject: [RHEL7.1 PATCH 09/82] raid5: fix stripe release order
Bugzilla: 1085530
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Shaohua Li <shli@kernel.org>

patch "make release_stripe lockless" changes the order stripes are released.
Originally I thought block layer can take care of request merge, but it appears
there are still some requests not merged. It's easy to fix the order.

Signed-off-by: Shaohua Li <shli@fusionio.com>
Signed-off-by: NeilBrown <neilb@suse.de>
(partial backport of d265d9dc1d25a69affc21ae9fe5004b9d09c10ef
 due to out of order backport of b89241e8cdb8321c20546d47645a9b65b58113b5)
Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/raid5.c b/drivers/md/raid5.c
index d38b076..1130083 100644
--- a/drivers/md/raid5.c
+++ b/drivers/md/raid5.c
@@ -247,6 +247,7 @@ static int release_stripe_list(struct r5conf *conf)
  struct llist_node *head;
 
  head = llist_del_all(&conf->released_stripes);
+ head = llist_reverse_order(head);
  while (head) {
   sh = llist_entry(head, struct stripe_head, release_list);
   head = llist_next(head);
-- 
1.7.1